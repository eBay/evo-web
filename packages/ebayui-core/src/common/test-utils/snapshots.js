import { render, prettyDOM } from "@marko/testing-library";
import { expect } from "vitest";
import visualHTML from "visual-html";
import { diffLines } from "diff";

function snapshotHTML() {
    return async (template, input) => {
        const { container } = await render(template, input);
        expect(prettyDOM(container)).toMatchSnapshot();
    };
}

async function diffServerSnapshots(template, input) {
    const { container } = await render(template, input);
    return async (afterTemplate, afterInput) => {
        const { container: afterContainer } = await render(
            afterTemplate,
            afterInput,
        );
        return diffOnlyChanges(container, afterContainer);
    };
}

function diffOnlyChanges(a, b) {
    const changes = diffLines(a, b);
    return changes
        .filter((part) => part.added || part.removed) // omit unchanged lines
        .map((part) => {
            const prefix = part.added ? "+ " : "- ";
            return part.value
                .split("\n")
                .filter((line) => line.trim() !== "") // clean up empty lines
                .map((line) => prefix + line)
                .join("\n");
        })
        .join("\n");
}
function diffHTML(beforeHTML) {
    const before = visualHTML(beforeHTML);
    return (afterHTML) => diffOnlyChanges(before, visualHTML(afterHTML));
}

export {
    snapshotHTML,
    diffServerSnapshots,
    diffHTML,
    diffOnlyChanges,
    visualHTML,
};
