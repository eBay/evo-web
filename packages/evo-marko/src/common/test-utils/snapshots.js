import { render } from "@marko/testing-library";
import { expect } from "vitest";

function snapshotHTML() {
  return async (template, input) => {
    const { container } = await render(template, input);
    expect(container).toMatchSnapshot();
  };
}

export { snapshotHTML };
