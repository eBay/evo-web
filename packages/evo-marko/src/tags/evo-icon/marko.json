{
  "taglib-id": "evo-marko-icons",
  "tags-dir": "./tags"
}
