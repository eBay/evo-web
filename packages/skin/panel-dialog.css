@import "./dist/panel-dialog/panel-dialog.css";
