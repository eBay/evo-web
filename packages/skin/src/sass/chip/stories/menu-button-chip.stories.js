export default { title: "Skin/Chip/Menu Button Chip" };

export const collapsed = () => `
<button class="chip chip--filter" aria-expanded="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const expanded = () => `
<button class="chip chip--filter" aria-expanded="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" aria-expanded="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-chevron-down-12"/>
        </svg>
    </button>
</div>
`;
