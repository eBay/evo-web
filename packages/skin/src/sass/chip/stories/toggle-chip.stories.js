export default { title: "Skin/Chip/Toggle Chip" };

export const button = () => `
<button class="chip" type="button" aria-pressed="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const buttonSelected = () => `
<button class="chip" type="button" aria-pressed="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const anchor = () => `
<a class="chip" herf="https://www.ebay.com">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--selected" herf="https://www.ebay.com">
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip" type="button" aria-pressed="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-close-12"></use>
        </svg>
    </button>
</div>
`;
