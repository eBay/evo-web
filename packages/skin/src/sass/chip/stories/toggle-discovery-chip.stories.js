export default { title: "Skin/Chip/Toggle Discovery Chip" };

export const button = () => `
<button class="chip chip--discovery" type="button" aria-pressed="false">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const buttonSelected = () => `
<button class="chip chip--discovery" type="button" aria-pressed="true">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const anchor = () => `
<a class="chip chip--discovery" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--discovery chip--selected" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" type="button" aria-pressed="false">
        <img
            class="chip__media"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
            alt=""
        >
        <span class="chip__text">Football</span>
    </button>
</div>
`;
