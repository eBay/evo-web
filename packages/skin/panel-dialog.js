require('./dist/panel-dialog/panel-dialog.css');
