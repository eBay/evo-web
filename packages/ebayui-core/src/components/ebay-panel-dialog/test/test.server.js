import { describe, it } from "vitest";

import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import { testPassThroughAttributes } from "../../../common/test-utils/server";
import * as stories from "../panel-dialog.stories"; // import all stories from the stories file
const { Default, WithPrevButton, WithFooter } = composeStories(stories);
const htmlSnap = snapshotHTML(__dirname);

describe("dialog", () => {
    it("renders basic version", async () => {
        await htmlSnap(Default);
    });

    it("renders with previous button", async () => {
        await htmlSnap(WithPrevButton);
    });

    it("renders with footer", async () => {
        await htmlSnap(WithFooter);
    });

    testPassThroughAttributes(Default);
});
