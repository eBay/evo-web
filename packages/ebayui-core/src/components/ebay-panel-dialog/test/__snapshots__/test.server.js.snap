// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dialog > passes through additional html attributes 1`] = `
<div
  aria-labelledby="s0-0-0-dialog-title"
  aria-modal="true"
  class="panel-dialog class1 panel-dialog--mask-fade-slow"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  hidden=""
  role="dialog"
  style="color:red"
  type="number"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#13-->
      <h2
        class="panel-dialog__title"
        id="s0-0-0-dialog-title"
      >
        <!--F#2-->
        Heading
        <!--F/-->
      </h2>
      <!--F/-->
      <!--F#14-->
      <button
        aria-label="Close Panel"
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      <!--F#1-->
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        <a
          href="http://www.ebay.com"
        >
          www.ebay.com
        </a>
      </p>
      <!--F/-->
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > passes through additional html attributes 2`] = `
<div
  aria-labelledby="s0-0-0-dialog-title"
  aria-modal="true"
  class="panel-dialog class1 panel-dialog--mask-fade-slow"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  hidden=""
  role="dialog"
  style="color:red"
  type="number"
>
  <div
    class="panel-dialog__window panel-dialog__window--slide"
  >
    <div
      class="panel-dialog__header"
    >
      <!--F#13-->
      <h2
        class="panel-dialog__title"
        id="s0-0-0-dialog-title"
      >
        <!--F#2-->
        Heading
        <!--F/-->
      </h2>
      <!--F/-->
      <!--F#14-->
      <button
        aria-label="Close Panel"
        class="icon-btn panel-dialog__close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="icon icon--16"
          focusable="false"
        >
          <defs>
            <symbol
              id="icon-close-16"
              viewBox="0 0 16 16"
            >
              <path
                d="M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"
              />
            </symbol>
          </defs>
          <use
            href="#icon-close-16"
          />
        </svg>
      </button>
      <!--F/-->
    </div>
    <div
      class="panel-dialog__main"
    >
      <!--F#16-->
      <!--F#1-->
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
      <p>
        <a
          href="http://www.ebay.com"
        >
          www.ebay.com
        </a>
      </p>
      <!--F/-->
      <!--F/-->
    </div>
  </div>
</div>
`;

exports[`dialog > renders basic version 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33maria-labelledby[39m=[32m"s0-0-0-dialog-title"[39m
    [33maria-modal[39m=[32m"true"[39m
    [33mclass[39m=[32m"panel-dialog panel-dialog--mask-fade-slow"[39m
    [33mhidden[39m=[32m""[39m
    [33mrole[39m=[32m"dialog"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"panel-dialog__window panel-dialog__window--slide"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__header"[39m
      [36m>[39m
        [36m<h2[39m
          [33mclass[39m=[32m"panel-dialog__title"[39m
          [33mid[39m=[32m"s0-0-0-dialog-title"[39m
        [36m>[39m
          [0mHeading[0m
        [36m</h2>[39m
        [36m<button[39m
          [33maria-label[39m=[32m"Close Panel"[39m
          [33mclass[39m=[32m"icon-btn panel-dialog__close"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [36m<svg[39m
            [33maria-hidden[39m=[32m"true"[39m
            [33mclass[39m=[32m"icon icon--16"[39m
            [33mfocusable[39m=[32m"false"[39m
          [36m>[39m
            [36m<defs>[39m
              [36m<symbol[39m
                [33mid[39m=[32m"icon-close-16"[39m
                [33mviewBox[39m=[32m"0 0 16 16"[39m
              [36m>[39m
                [36m<path[39m
                  [33md[39m=[32m"M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"[39m
                [36m/>[39m
              [36m</symbol>[39m
            [36m</defs>[39m
            [36m<use[39m
              [33mhref[39m=[32m"#icon-close-16"[39m
            [36m/>[39m
          [36m</svg>[39m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__main"[39m
      [36m>[39m
        [36m<p>[39m
          [0mLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.[0m
        [36m</p>[39m
        [36m<p>[39m
          [36m<a[39m
            [33mhref[39m=[32m"http://www.ebay.com"[39m
          [36m>[39m
            [0mwww.ebay.com[0m
          [36m</a>[39m
        [36m</p>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
  [36m<button[39m
    [33mclass[39m=[32m"btn btn--secondary"[39m
    [33mdata-ebayui[39m=[32m""[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [0mOpen Dialog[0m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;

exports[`dialog > renders with footer 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33maria-labelledby[39m=[32m"s0-0-0-dialog-title"[39m
    [33maria-modal[39m=[32m"true"[39m
    [33mclass[39m=[32m"panel-dialog panel-dialog--mask-fade-slow"[39m
    [33mhidden[39m=[32m""[39m
    [33mrole[39m=[32m"dialog"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"panel-dialog__window panel-dialog__window--slide"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__header"[39m
      [36m>[39m
        [36m<h2[39m
          [33mclass[39m=[32m"panel-dialog__title"[39m
          [33mid[39m=[32m"s0-0-0-dialog-title"[39m
        [36m>[39m
          [0mHeading[0m
        [36m</h2>[39m
        [36m<button[39m
          [33maria-label[39m=[32m"Close Dialog"[39m
          [33mclass[39m=[32m"icon-btn panel-dialog__close"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [36m<svg[39m
            [33maria-hidden[39m=[32m"true"[39m
            [33mclass[39m=[32m"icon icon--16"[39m
            [33mfocusable[39m=[32m"false"[39m
          [36m>[39m
            [36m<defs>[39m
              [36m<symbol[39m
                [33mid[39m=[32m"icon-close-16"[39m
                [33mviewBox[39m=[32m"0 0 16 16"[39m
              [36m>[39m
                [36m<path[39m
                  [33md[39m=[32m"M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"[39m
                [36m/>[39m
              [36m</symbol>[39m
            [36m</defs>[39m
            [36m<use[39m
              [33mhref[39m=[32m"#icon-close-16"[39m
            [36m/>[39m
          [36m</svg>[39m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__main"[39m
      [36m>[39m
        [36m<p>[39m
          [0mLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.[0m
        [36m</p>[39m
        [36m<p>[39m
          [36m<a[39m
            [33mhref[39m=[32m"http://www.ebay.com"[39m
          [36m>[39m
            [0mwww.ebay.com[0m
          [36m</a>[39m
        [36m</p>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__footer"[39m
      [36m>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn"[39m
        [36m>[39m
          [0mButton 1[0m
        [36m</button>[39m
        [36m<button[39m
          [33mclass[39m=[32m"btn"[39m
        [36m>[39m
          [0mButton 2[0m
        [36m</button>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
  [36m<button[39m
    [33mclass[39m=[32m"btn btn--secondary"[39m
    [33mdata-ebayui[39m=[32m""[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [0mOpen Dialog[0m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;

exports[`dialog > renders with previous button 1`] = `
"[36m<DocumentFragment>[39m
  [36m<div[39m
    [33maria-labelledby[39m=[32m"s0-0-0-dialog-title"[39m
    [33maria-modal[39m=[32m"true"[39m
    [33mclass[39m=[32m"panel-dialog panel-dialog--mask-fade-slow"[39m
    [33mhidden[39m=[32m""[39m
    [33mrole[39m=[32m"dialog"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"panel-dialog__window panel-dialog__window--slide"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__header"[39m
      [36m>[39m
        [36m<button[39m
          [33maria-label[39m=[32m"Go back"[39m
          [33mclass[39m=[32m"icon-btn panel-dialog__prev"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [36m<svg[39m
            [33maria-hidden[39m=[32m"true"[39m
            [33mclass[39m=[32m"icon icon--16"[39m
            [33mfocusable[39m=[32m"false"[39m
          [36m>[39m
            [36m<defs>[39m
              [36m<symbol[39m
                [33mid[39m=[32m"icon-chevron-left-16"[39m
                [33mviewBox[39m=[32m"0 0 16 16"[39m
              [36m>[39m
                [36m<path[39m
                  [33md[39m=[32m"M3.293 8.708a1 1 0 0 1 0-1.415l6-6a1 1 0 0 1 1.414 1.414L5.414 8l5.293 5.293a1 1 0 0 1-1.414 1.414l-6-6Z"[39m
                [36m/>[39m
              [36m</symbol>[39m
            [36m</defs>[39m
            [36m<use[39m
              [33mhref[39m=[32m"#icon-chevron-left-16"[39m
            [36m/>[39m
          [36m</svg>[39m
        [36m</button>[39m
        [36m<h2[39m
          [33mclass[39m=[32m"panel-dialog__title"[39m
          [33mid[39m=[32m"s0-0-0-dialog-title"[39m
        [36m>[39m
          [0mHeading[0m
        [36m</h2>[39m
        [36m<button[39m
          [33maria-label[39m=[32m"Close Dialog"[39m
          [33mclass[39m=[32m"icon-btn panel-dialog__close"[39m
          [33mtype[39m=[32m"button"[39m
        [36m>[39m
          [36m<svg[39m
            [33maria-hidden[39m=[32m"true"[39m
            [33mclass[39m=[32m"icon icon--16"[39m
            [33mfocusable[39m=[32m"false"[39m
          [36m>[39m
            [36m<defs>[39m
              [36m<symbol[39m
                [33mid[39m=[32m"icon-close-16"[39m
                [33mviewBox[39m=[32m"0 0 16 16"[39m
              [36m>[39m
                [36m<path[39m
                  [33md[39m=[32m"M2.293 2.293a1 1 0 0 1 1.414 0L8 6.586l4.293-4.293a1 1 0 1 1 1.414 1.414L9.414 8l4.293 4.293a1 1 0 0 1-1.414 1.414L8 9.414l-4.293 4.293a1 1 0 0 1-1.414-1.414L6.586 8 2.293 3.707a1 1 0 0 1 0-1.414Z"[39m
                [36m/>[39m
              [36m</symbol>[39m
            [36m</defs>[39m
            [36m<use[39m
              [33mhref[39m=[32m"#icon-close-16"[39m
            [36m/>[39m
          [36m</svg>[39m
        [36m</button>[39m
      [36m</div>[39m
      [36m<div[39m
        [33mclass[39m=[32m"panel-dialog__main"[39m
      [36m>[39m
        [36m<p>[39m
          [0mLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.[0m
        [36m</p>[39m
        [36m<p>[39m
          [36m<a[39m
            [33mhref[39m=[32m"http://www.ebay.com"[39m
          [36m>[39m
            [0mwww.ebay.com[0m
          [36m</a>[39m
        [36m</p>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
  [36m<button[39m
    [33mclass[39m=[32m"btn btn--secondary"[39m
    [33mdata-ebayui[39m=[32m""[39m
    [33mtype[39m=[32m"button"[39m
  [36m>[39m
    [0mOpen Dialog[0m
  [36m</button>[39m
[36m</DocumentFragment>[39m"
`;
